rvps/src/extractors/extractor_modules/mod.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Add your specific provenance declaration here.
+
+use anyhow::*;
+use std::collections::HashMap;
+
+use crate::reference_value::ReferenceValue;
+
+/// Extractor is a standard interface that all provenance extractors
+/// need to implement. Here reference_value can be modified in the
+/// handler, added any field if needed.
+pub trait Extractor {
+    fn verify_and_extract(&self, provenance: &str) -> Result<ReferenceValue>;
```

Comment:
So one piece of provenance will always result in just one reference value? There's no way to have some kind of manifest that accounts for many different components?

Reply:
Wow, you are right. Let me update this as `Result<Vec<ReferenceValue>>`